pyhf/tensor/tensorflow_backend.py

```diff
@@ -414,3 +414,26 @@ def normal_cdf(self, x, mu=0, sigma=1):
         sigma = self.astensor(sigma)
         normal = tfp.distributions.Normal(mu, sigma)
         return normal.cdf(x)
+
+    class Poisson(object):
+        def __init__(self, rate):
+            self.pdf =  tfp.distributions.Poisson(rate = rate)
+            self.batch_shape = self.pdf.batch_shape
+        def sample(self, *args, **kwargs):
+            return self.pdf.sample(*args, **kwargs)
+        
+        def log_prob(self, data):
+            d = self.pdf.log_prob(data)
```

Comment:
In 

`d = self.pdf.log_prob(data)`

this `d` never gets used. That doesn't seem right.